### PR TITLE
Remove `Message` and add support of any `MessageInterface` implementation

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️

Linked PR: https://github.com/yiisoft/mailer/pull/84

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -33,9 +33,10 @@
         "psr/event-dispatcher": "^1.0",
         "symfony/mailer": "^5.3|^6.0|^7.0",
         "symfony/mime": "^5.4|^6.2|^7.0",
-        "yiisoft/mailer": "^5.0"
+        "yiisoft/mailer": "dev-big-refactoring as 5.1"
```
Comment (arogachev) [inline composer.json:36]: Change to stable version before merge.

**Review by arogachev — APPROVED** [review]

File: src/Mailer.php [hunk 2]
```diff
@@ -23,6 +23,8 @@
  * Mailer implements a mailer based on Symfony Mailer.
  *
  * @see https://github.com/symfony/mailer
+ *
+ * @api
```
Comment (samdark) [inline src/Mailer.php:27]: ?

**Review by samdark — APPROVED** [review]
Code looks alright.

File: src/Mailer.php (hunk 2 above)
Comment (vjik) [reply src/Mailer.php:27]: See:
- https://psalm.dev/docs/annotating_code/supported_annotations/#api-psalm-api
- https://psalm.dev/docs/running_psalm/configuration/#findunusedcode

**Pull request merged**